src/lib/registration/clients/scc_auto.rb

```diff
@@ -0,0 +1,409 @@
+          options.email = @config.email
+          options.reg_code = @config.reg_code
+
+          registration_ui.register_system_and_base_product
+        end
+      end
+
+      # register the addons specified in the profile
+      def register_addons
+        # set the option for installing the updates for addons
+        options = Registration::Storage::InstallationOptions.instance
+        options.install_updates = @config.install_updates
+
+        ay_addons_handler = Registration::AutoyastAddons.new(@config.addons, registration)
+        ay_addons_handler.select
+        ay_addons_handler.register
+
+        # select the new products to install
+        ::Registration::SwMgmt.select_addon_products
+      end
+
+      # was the system already registered?
+      # @return [Boolean] true if the system was alreay registered
+      def old_system_registered?
+        ::Registration::SwMgmt.copy_old_credentials(Yast::Installation.destdir)
+
+        # update the registration using the old credentials
+        ::File.exist?(SUSE::Connect::YaST::GLOBAL_CREDENTIALS_FILE)
+      end
+
+      # update the system registration
+      # @return [Boolean] true on success
+      def update_system_registration
+        registration_ui.update_system
+      end
+
+      # @yieldreturn [Boolean, SUSE::Connect::Remote::Product] success flag and
+      #   remote product pair
+      # @return [Boolean] true on success
+      def handle_product_service(&block)
+        success, product_service = block.call
+        return false unless success
+
+        # keep updates enabled?
+        return true if @config.install_updates || !product_service
+
+        registration_ui.disable_update_repos(product_service)
+      end
+
+      # migrate registration if applicable or skip or report issue.
+      def migrate_reg
+        if old_system_registered?
+          # act always like we have online only repo for registered system
+          Y2Packager::MediumType.type = :online
+
+          # drop all obsolete repositories and services (manual upgrade contains a dialog
+          # where the old repositories are deleted, in AY we need to do it automatically here)
+          # Note: the Update module creates automatically a backup which is restored
+          # when upgrade is aborted or crashes.
+          repo_cleanup
+
+          ret = ::Registration::UI::OfflineMigrationWorkflow.new.main
+          log.info "Migration result: #{ret}"
+          ret == :next
+        # Full medium we can upgrade without registration
+        elsif Y2Packager::MediumType.offline?
+          true
+        else
+          # Intentionally use blocking popup as it is fatal error that stops installation.
+          Yast::Popup.Error(
+            # TRANSLATORS: profile wants to do registration, but old system is not registered.
+            _("Old system is not registered and autoyast profile require registration.\n" \
```

Comment:
We should improve it when already touching it:

```suggestion
            _("The old system is not registered and the AutoYaST profile requires registration.\n" \
```